mlflow/server/auth/__init__.py

```diff
@@ -585,6 +677,80 @@ def filter_search_experiments(resp: Response):
     resp.data = message_to_json(response_message)
 
 
+def filter_search_logged_models(resp: Response) -> None:
+    """
+    Filter out unreadable logged models from the search results.
+    """
+    from mlflow.utils.search_utils import SearchLoggedModelsPaginationToken as Token
+
+    if sender_is_admin():
+        return
+
+    response_proto = SearchLoggedModels.Response()
+    parse_dict(resp.json, response_proto)
+
+    # fetch permissions
+    username = authenticate_request().username
+    perms = store.list_experiment_permissions(username)
```

Comment:
This might be inefficient if there're lots of experiments, but maybe fine for now 🤔

Reply:
It's inefficient. I don't think this approach is gonna scale, to be honest.